embassy-net-adin1110/src/lib.rs

```diff
@@ -0,0 +1,1249 @@
+#![allow(clippy::pedantic)]
+#![feature(type_alias_impl_trait)]
+#![feature(async_fn_in_trait)]
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+
+use ch::driver::LinkState;
+use embassy_futures::select::{select, Either};
+use embassy_net_driver_channel as ch;
+use embassy_time::{Duration, Timer};
+use embedded_hal_1::digital::OutputPin;
+use embedded_hal_async::digital::Wait;
+use embedded_hal_async::spi::{Operation, SpiDevice};
+use heapless::Vec;
+
+mod crc32;
+mod crc8;
+mod mdio;
+mod phy;
+mod regs;
+
+pub use crc32::ETH_FSC;
+use crc8::crc8;
+pub use mdio::MdioBus;
+pub use phy::{Phy10BaseT1x, RegsC22, RegsC45};
+pub use regs::{Config0, Config2, SpiRegisters as sr, Status0, Status1};
+
+use crate::regs::{LedCntrl, LedFunc, LedPol, LedPolarity, SpiHeader};
+
+pub const PHYID: u32 = 0x0283BC91;
+
+/// Error values ADIN1110
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[allow(non_camel_case_types)]
+pub enum AdinError<E> {
+    Spi(E),
+    SENDERROR,
+    READERROR,
+    CRC,
+    PACKET_TOO_BIG,
+    PACKET_TOO_SMALL,
+    MDIO_ACC_TIMEOUT,
+}
+
+pub type AEResult<T, SPIE> = core::result::Result<T, AdinError<SPIE>>;
+pub const MDIO_PHY_ADDR: u8 = 0x01;
+pub const MTU: usize = 1500;
```

Comment:
ethernet MTU is usually 1514 (14 bytes ethernet header + 1500 bytes IP payload)

Reply:
Is fixed